app/src/pipelines/email_result/pipeline_wrapper.py

```diff
@@ -28,34 +30,46 @@ def setup(self) -> None:
 
     def run_api(self, result_id: str, email: str) -> dict:
         with using_metadata({"email": email}):
-            try:
-                response = self.pipeline.run(
-                    {
-                        "logger": {
-                            "messages_list": [{"result_id": result_id, "email": email}],
-                        },
-                        "load_result": {
-                            "result_id": result_id,
-                        },
-                        "email_result": {
-                            "email": email,
-                        },
+            # Must set using_metadata context before calling tracer.start_as_current_span()
+            assert isinstance(tracer, _tracers.OITracer), f"Got unexpected {type(tracer)}"
+            with tracer.start_as_current_span(  # pylint: disable=not-context-manager,unexpected-keyword-arg
+                self.name, openinference_span_kind="chain"
+            ) as span:
+                result = self._run(result_id, email)
+                span.set_input(result_id)
+                span.set_output(result["email_result"]["status"])
```

Comment:
More control over the `input` and `output` columns in Phoenix's table view of the traces